Code/Framework/AzCore/AzCore/DOM/DomPath.cpp

```diff
@@ -0,0 +1,455 @@
+        return formattedString;
+    }
+
+    void Path::FromString(AZStd::string_view pathString)
+    {
+        m_entries.clear();
+        size_t pathIndex = 0;
+        bool isNumber = true;
+        for (size_t i = 0; i <= pathString.size(); ++i)
+        {
+            if (i == pathString.size() || pathString[i] == PathSeparator)
+            {
+                if (i > 0)
+                {
+                    AZStd::string_view section(&pathString.data()[pathIndex], i - pathIndex);
+                    if (section.size() == 1 && section[0] == EndOfArrayCharacter)
+                    {
+                        PathEntry entry;
+                        entry.SetEndOfArray();
+                        m_entries.push_back(AZStd::move(entry));
+                    }
+                    else if (isNumber && section.size() > 0)
+                    {
+                        size_t index = 0;
+                        ConsoleTypeHelpers::StringToValue(index, section);
+                        m_entries.push_back(PathEntry{ index });
+                    }
+                    else
+                    {
+                        AZStd::string convertedSection;
+                        size_t lastPos = 0;
+                        size_t posToEscape = section.find(EscapeCharacter);
+                        while (posToEscape != AZStd::string_view::npos)
+                        {
+                            if (convertedSection.empty())
+                            {
+                                convertedSection.reserve(section.size() - 1);
+                            }
+                            convertedSection += section.substr(lastPos, posToEscape - lastPos);
+                            if (section[posToEscape + 1] == ForwardSlashSequence)
+                            {
+                                convertedSection += '/';
+                            }
+                            else
+                            {
+                                convertedSection += '~';
+                            }
+
+                            lastPos = posToEscape + 2;
+                            posToEscape = section.find(EscapeCharacter, posToEscape + 2);
+                        }
+
+                        if (!convertedSection.empty())
+                        {
+                            convertedSection += section.substr(lastPos);
+                            m_entries.push_back(PathEntry{ convertedSection });
+                        }
+                        else
+                        {
+                            m_entries.push_back(PathEntry{ section });
+                        }
+                    }
+                }
+                pathIndex = i + 1;
+                isNumber = true;
+                continue;
+            }
+
+            const char c = pathString[i];
+            if (isNumber && !(c >= '0' && c <= '9'))
+            {
+                isNumber = false;
+            }
```

Comment:
This could just set isNumber to the result of the boolean expression instead of having a conditional.

```suggestion
            isNumber = pathString[i] >= '0' && pathString[i] <= 0;
```

Reply:
Rolled it into a single assignment.